### PR TITLE
[SPARK-49511][SQL] Apply formatting rules to sql/api

### PR DESCRIPTION
**Pull request by hvanhovell**
### What changes were proposed in this pull request?

This PR proposes to apply formatting rules to sql/api

### Why are the changes needed?

The package is new, and we can start autoformatting. It won't have downside of backporting difficulty and etc.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Yes, by `dev/lint-scala`

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

**Comment by hvanhovell** [comment]
cc @HyukjinKwon we discussed this.

File: sql/api/src/main/scala/org/apache/spark/sql/Column.scala [hunk 1]
```diff
@@ -168,23 +168,20 @@ class Column(val node: ColumnNode) extends Logging {
...
+   * into the correct JVM types.
    * @since 1.6.0
    */
-  def as[U : Encoder]: TypedColumn[Any, U] = new TypedColumn[Any, U](node, implicitly[Encoder[U]])
+  def as[U: Encoder]: TypedColumn[Any, U] = new TypedColumn[Any, U](node, implicitly[Encoder[U]])
 
   /**
-   * Extracts a value or values from a complex type.
-   * The following types of extraction are supported:
-   * <ul>
```
Comment (hvanhovell) [inline sql/api/src/main/scala/org/apache/spark/sql/Column.scala:181]: I am not sure if we want this.

File: sql/api/src/main/scala/org/apache/spark/sql/Column.scala [hunk 2]
```diff
@@ -765,13 +761,11 @@ class Column(val node: ColumnNode) extends Logging {
    * A boolean expression that is evaluated to true if the value of this expression is contained
    * by the evaluated values of the arguments.
    *
-   * Note: Since the type of the elements in the list are inferred only during the run time,
-   * the elements will be "up-casted" to the most common type for comparison.
-   * For eg:
```
Comment (hvanhovell) [inline sql/api/src/main/scala/org/apache/spark/sql/Column.scala:770]: List is messed up...

**Comment by dongjoon-hyun** [comment]
Thank you for the clean-ups.

**Review by HyukjinKwon — APPROVED** [review]

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Manually tested.

Merged to master.

**Pull request closed without merging**